public/static-checkout/js/scripts.js

```diff
@@ -0,0 +1,309 @@
+}
+
+function redirectToMFE(couponCode) {
+  var queryString = window.location.search;
+  var urlParams = new URLSearchParams(queryString);
+  var sku = urlParams.get('sku');
+  var newUrl = ecommerceBaseUrl + '/basket/add/?sku=' + sku;
+  if (couponCode) {
+    newUrl = newUrl.concat('&coupon=', couponCode);
+  }
+  window.location.href = newUrl;
+}
+
+userButton.addEventListener('click', function toggleUserDropdown() {
+  document.querySelector('.dropdown-menu').classList.toggle('show');
+}, false);
+
+window.onclick = function closeMenu(event) {
+  if (!event.target.matches('.user-button')) {
+    var dropdownMenu = document.getElementsByClassName('dropdown-menu');
+    var i;
+    for (i = 0; i < dropdownMenu.length; i += 1) {
+      var open = dropdownMenu[i];
+      if (open.classList.contains('show')) {
+        open.classList.remove('show');
+      }
+    }
+  }
+};
+
+function renderStates(country) {
+  var states = Object.values(countryStates[country]);
+  for (var i = 0; i < states.length; i += 1) {
+    var state = document.createElement('option');
+    state.innerHTML = states[i];
+    stateSelect.appendChild(state);
+  }
+}
+
+function renderCountryState() {
+  countrySelect.onchange = function getCountryStates() {
+    var stateSelectHtml = document.createElement('div');
+    stateSelectHtml.innerHTML = '<select id="state" name="state"><option value="" selected="selected">Choose state/province</option></select>';
+    stateSelect.length = 1;
+    if (!(countrySelect.value === 'CA' || countrySelect.value === 'US')) {
+      stateSelectHtml = document.createElement('div');
+      stateSelectHtml.innerHTML = '<input id="state" type="text" name="state">';
+      stateSelect.parentNode.replaceChild(stateSelectHtml, stateSelect);
+      stateSelect = document.getElementById('state');
+    } else {
+      stateSelect.parentNode.replaceChild(stateSelectHtml, stateSelect);
+      stateSelect = document.getElementById('state');
+      if (countrySelect.value === 'CA') {
+        renderStates('CA');
+      } else if (countrySelect.value === 'US') {
+        renderStates('US');
+      }
+    }
+  };
+}
+
+document.addEventListener('DOMContentLoaded', function handleCountrySelection() {
+  renderCountryState();
+});
+
+couponButton.addEventListener('click', function couponRedirectToMicrofrontend(event) {
+  event.preventDefault();
+  var couponCode = document.getElementById('couponcode').value;
+  redirectToMFE(couponCode);
+}, false);
+
+
+function parseCookies() {
+  var cookies = {};
+  if (document.cookie && document.cookie !== '') {
+    document.cookie.split(';').forEach(function (c) {
+      var m = c.trim().match(/(\w+)=(.*)/);
+      if (m !== undefined) {
+        cookies[m[1]] = decodeURIComponent(m[2]);
+      }
+    });
+  }
+  return cookies;
+}
+
+function disableForm() {
+  var toDisable = document.querySelectorAll('input, select, button');
+  Array.prototype.forEach.call(toDisable, function (elementToDisable) {
```

Comment:
Again, may wanna watch the `forEach` - it's also generally slower than a normal for loop, I think.